### PR TITLE
fix: replaced className with data attribute

### LINKED ISSUE #1152: ✨ [Feature request]: Add data-attribute for Umami tracking
### Is your feature request related to a problem? Please describe.

We have integrated ReactPlay with [Umami](https://umami.is/) for tracking analytics. 

### Describe the solution you'd like.

We need to add data-attribute to components that are to be tracked on the umami analytics. Previously they were maintained as className.

### Describe alternatives you've considered.

- Refer to https://umami.is/docs/track-events to understand how the appropriate data-attributes can be added.

### Screenshot / Screenshare

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by supminn**
# Description

Replaced umami related class names with data attribute values.

- Fixes #1152

### CODE REVIEW

File: src/common/activities/ActivityBanner.jsx [hunk 1]
```diff
@@ -39,7 +39,8 @@ function ActivityBanner({ currentActivity }) {
               <span className="btn-label">Get started</span>
             </a>
             <a
-              className="body-c2a-btn umami--click--github"
+              className="body-c2a-btn "
+              data-umami-event="gitub-repo-link"
```
Comment (Sachin-chaurasiya) [inline src/common/activities/ActivityBanner.jsx:43]:
```suggestion
              data-umami-event="github-button"
```

File: src/common/defaultBanner/DefaultBanner.jsx [hunk 2]
```diff
@@ -24,7 +24,8 @@ const DefaultBanner = () => {
           <span className="btn-label">Browse</span>
         </Link>
         <a
-          className="body-c2a-btn umami--click--github"
+          className="body-c2a-btn"
+          data-umami-event="github-repo-link"
```
Comment (Sachin-chaurasiya) [inline src/common/defaultBanner/DefaultBanner.jsx:28]:
```suggestion
          data-umami-event="github-button"
```

File: src/common/header/HeaderNav.jsx [hunk 3]
```diff
@@ -99,7 +99,8 @@ const HeaderNav = ({ showBrowse }) => {
               </a>
             ) : (
               <a
-                className="app-header-btn app-header-btn--primary umami--click--create-play"
+                className="app-header-btn app-header-btn--primary"
+                data-umami-event="create-play"
```
Comment (Sachin-chaurasiya) [inline src/common/header/HeaderNav.jsx:103]:
```suggestion
                data-umami-event="create-play-button"
```

File: src/common/header/HeaderNav.jsx [hunk 4]
```diff
@@ -134,7 +135,8 @@ const HeaderNav = ({ showBrowse }) => {
           </li>
           <li>
             <a
-              className="app-header-btn app-header-btn--default umami--click--github"
+              className="app-header-btn app-header-btn--default"
+              data-umami-event="github-repo-link"
```
Comment (Sachin-chaurasiya) [inline src/common/header/HeaderNav.jsx:139]:
```suggestion
              data-umami-event="github-button"
```

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Left a few comments, thanks

File: src/common/activities/ActivityBanner.jsx [hunk 5]
```diff
@@ -40,7 +40,7 @@ function ActivityBanner({ currentActivity }) {
             </a>
             <a
               className="body-c2a-btn "
-              data-umami-event="gitub-repo-link"
+              data-umami-event="gitub-button"
```
Comment (Sachin-chaurasiya) [inline src/common/activities/ActivityBanner.jsx:43]:
```suggestion
              data-umami-event="github-button"
```

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]

**Comment by Sachin-chaurasiya** [comment]
@supminn , I was thinking to have all the event's names in a constant, so that we will have a single source of truth, and whenever we want to make any changes we have to do that in one place only. wdyt?

```js
const UMAMI_EVENTS = {
GITHUB_BUTTON : "github-button",
CREATE_PLAY_BUTTON : "create-play-button"
}
```

**Comment by supminn** [comment]
> @supminn , I was thinking to have all the event's names in a constant, so that we will have a single source of truth, and whenever we want to make any changes we have to do that in one place only. wdyt?
> 
> ```js
> const UMAMI_EVENTS = {
> GITHUB_BUTTON : "github-button",
> CREATE_PLAY_BUTTON : "create-play-button"
> }
> ```

This is a great suggestion. I'll get these changes done..
Do you think the events should be named `github-button` or `github-click`? Since this value would be displayed on the Umami dashboard as an event action.

**Comment by Sachin-chaurasiya** [comment]
> > @supminn , I was thinking to have all the event's names in a constant, so that we will have a single source of truth, and whenever we want to make any changes we have to do that in one place only. wdyt?
> > ```js
> > const UMAMI_EVENTS = {
> > GITHUB_BUTTON : "github-button",
> > CREATE_PLAY_BUTTON : "create-play-button"
> > }
> > ```
> 
> This is a great suggestion. I'll get these changes done.. Do you think the events should be named `github-button` or `github-click`? Since this value would be displayed on the Umami dashboard as an event action.

We can keep the `github-button`.

**Review by Sachin-chaurasiya — APPROVED** [review]
Great work @supminn 🙌🏻,
Thanks for the contributions ✨

**Comment by Sachin-chaurasiya** [comment]
@atapas , Please review and merge.

**Review by atapas — APPROVED** [review]

**Comment by atapas** [comment]
@supminn Build fails

image

**Review by atapas — APPROVED** [review]

**Pull request merged**